services/scan/src/precinct_scanner_app.ts

```diff
@@ -366,11 +404,26 @@ export async function buildPrecinctScannerApp(
         return;
       }
 
-      store.setMarkThresholdOverrides(
-        bodyParseResult.ok().markThresholdOverrides
-      );
-      await updateInterpreterConfig(interpreter, workspace);
+      // Start a new batch
+      const ongoingBatchId = store.getOngoingBatchId();
+      assert(typeof ongoingBatchId === 'string');
```

Comment:
Thinking about this... If this assertion fails, it really is a failure of server state and an "Internal Server Error" rather than a "Bad Request." But if this function throws an error and crashes, does the whole backend crash?

Reply:
Based on how we are managing state on the server, the assertion _should_ be true and it really is a "Internal Server Error" rather than a "Bad Request." So I actually think the current behavior is appropriate? There's not a request I can think of that would trigger the assertion to fail, only changes to server logic.